src/v0/destinations/klaviyo/util.js

```diff
@@ -1,116 +1,117 @@
+  defaultBatchRequestConfig,
+  getSuccessRespEvents,
+  defaultPatchRequestConfig,
 } = require('../../util');
 
-const { BASE_ENDPOINT, LIST_CONF, MAPPING_CONFIG, CONFIG_CATEGORIES } = require('./config');
-const {
-  getDynamicErrorType,
-  processAxiosResponse,
-} = require('../../../adapters/utils/networkUtils');
+const { BASE_ENDPOINT, MAPPING_CONFIG, CONFIG_CATEGORIES, MAX_BATCH_SIZE } = require('./config');
+const { JSON_MIME_TYPE } = require('../../util/constant');
 const { NetworkError } = require('../../util/errorTypes');
+const { getDynamicErrorType } = require('../../../adapters/utils/networkUtils');
 const tags = require('../../util/tags');
-const { JSON_MIME_TYPE } = require('../../util/constant');
-
-/**
- * This function is used to check if the user/profile already exists or not, if already exists unique person_id
- * for that user is getting returned else false is returned.
- * Docs: https://developers.klaviyo.com/en/reference/get-profile-id
- * @param {*} message
- * @param {*} Config
- * @returns
- */
-const isProfileExist = async (message, { Config }) => {
-  const { privateApiKey } = Config;
-  const userIdentifiers = {
-    email: getFieldValueFromMessage(message, 'email'),
-    external_id: getFieldValueFromMessage(message, 'userId'),
-    phone_number: getFieldValueFromMessage(message, 'phone'),
-  };
+const { handleHttpRequest } = require('../../../adapters/network');
 
-  // eslint-disable-next-line no-restricted-syntax
-  for (const id in userIdentifiers) {
-    if (isDefinedAndNotNull(userIdentifiers[id])) {
-      // eslint-disable-next-line no-await-in-loop
-      const profileResponse = await httpGET(`${BASE_ENDPOINT}/api/v2/people/search`, {
-        header: {
-          Accept: JSON_MIME_TYPE,
-        },
-        params: {
-          api_key: privateApiKey,
-          [id]: userIdentifiers[id],
-        },
-      });
-      const processedProfileResponse = processAxiosResponse(profileResponse);
-      if (processedProfileResponse.status === 200 && processedProfileResponse.response?.id) {
-        return processedProfileResponse.response.id;
-      }
-      if (
-        !(
-          processedProfileResponse.status === 404 &&
-          processedProfileResponse?.response.detail ===
-            'There is no profile matching the given parameters.'
-        )
-      ) {
-        throw new NetworkError(
-          `The lookup call could not be completed with the error:
-          ${JSON.stringify(processedProfileResponse.response)}`,
-          processedProfileResponse.status,
-          {
-            [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedProfileResponse.status),
-          },
-          processedProfileResponse,
-        );
-      }
-    }
+const getProfileId = async (endpoint, payload, requestOptions) => {
+  let profileId;
+  const { httpResponse: resp } = await handleHttpRequest('post', endpoint, payload, requestOptions);
+  if (resp.response?.status === 201) {
```

Comment:
Couple of things on response handling
1. Does the API not respond with `200` ?
2. `resp.response?.response?.status` what would happen to Ln 34 ? -- will the event be aborted ?!
3. Would the klaviyo API respond with an object in the response body ? -- If yes, how will that appear in the message ``Failed to create user due to ${resp.response?.data}``